### ISSUE TITLE
[Request] Add special NBT tag

### ISSUE THREAD

**Issue by RainOfPain125** [mcMMO-Dev/mcMMO#3964]
People love automation, especially automated smelting with the use of hoppers. But then smelting skills aren't involved. Thats why I suggest everytime a player inserts a smelt-able ore into a container, it adds a special NBT tag with that players username, so that when it gets to the furnace it points to that players smelt skill, and uses whatever benefits it would give.

**Comment by QuantumToasted** [comment]
I can get behind this, assuming it's not already slated. It sucks that players are hindered from making automated furnaces and farms because it effectively prevents you from leveling up the associated skill, even if you go through the effort to build it. I made a huge farm so i can grind herbalism but it sucks that i can't automate it at all, because then i would've been able to make the farm more compact.

**Comment by RainOfPain125** [comment]
Yeah I agree @QuantumToasted . I'm not a coder, but I can assume this can be done if they wanted it to be done. I also think the same could be done with farms automated by water

Flowing water destroys crop, McMMO finds who that flowing water's source block was from, and apply their herbal-ism skills and their upgrades. If it comes from a dispenser, then whoever placed the dispenser is the target. I dont think it would get too resource intensive, and if it were a little, I think they could easily add a true/false option for automated systems such as the ones we've suggested here in case server owners have the fear of lag.

**Comment by nossr50** [comment]
Yeah this can be done, I'll be doing something quite similar to this, not sure if it would happen in 2.2 or a patch afterwards
